dotnet/src/webdriver/Internal/Logging/LogContextManager.cs

```diff
@@ -17,45 +17,31 @@
 // under the License.
 // </copyright>
 
+using System.Diagnostics.CodeAnalysis;
 using System.Threading;
 
+#nullable enable
+
 namespace OpenQA.Selenium.Internal.Logging
 {
     internal class LogContextManager
     {
-        private readonly ILogContext _globalLogContext;
-
-        private readonly AsyncLocal<ILogContext> _currentAmbientLogContext = new AsyncLocal<ILogContext>();
+        private readonly AsyncLocal<ILogContext?> _currentAmbientLogContext = new AsyncLocal<ILogContext?>();
 
         public LogContextManager()
         {
             var defaulConsoleLogHandler = new ConsoleLogHandler();
 
-            _globalLogContext = new LogContext(LogEventLevel.Info, null, null, new[] { defaulConsoleLogHandler });
+            GlobalContext = new LogContext(LogEventLevel.Info, null, null, new[] { defaulConsoleLogHandler });
         }
 
-        public ILogContext GlobalContext
-        {
-            get { return _globalLogContext; }
-        }
+        public ILogContext GlobalContext { get; }
 
+        [AllowNull]
```

Comment:
It cannot be `null`, at least by design. Can we adjust?

Reply:
The getter cannot return `null`, but the setter can be assigned `null`. That is what the `[AllowNull]` attribute is designed to handle.

Reply:
I cannot use to reversed mind, you will be our expert in this area!